### PR TITLE
Fix failed to push docker images with a specified tag

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <talonwan@yunify.com>

### What type of PR is this?

/kind flake

### What this PR does / why we need it:

Fix failed to push image with a wrong image tag. Whether the tag or a branch changes, the image should be pushed.

error logs: https://github.com/kubesphere/kubesphere/runs/2950932182?check_suite_focus=true#step:11:100

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -64,4 +64,4 @@ jobs:
         if: github.event_name == 'push'
         run: |
           echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
-          REPO=kubespheredev TAG=${GITHUB_REF#refs/heads/} make container-push
+          REPO=kubespheredev TAG=${GITHUB_REF#refs/*/} make container-push
```
Comment (yuswift) [inline .github/workflows/build.yml:67]: please modify https://github.com/kubesphere/kubesphere/blob/master/.github/workflows/build-multiarch.yaml too and cherry-pick this pr

Comment (yuswift) [reply .github/workflows/build.yml:67]: BTW, what's the value of `TAG ` if we run the workflow in tag `v3.1.1-rc.1`?

Comment (zheng1) [reply .github/workflows/build.yml:67]:
```bash
GITHUB_REF='refs/tags/v3.1.1-rc.1'
echo ${GITHUB_REF#refs/*/}
```

Comment (wansir) [reply .github/workflows/build.yml:67]: > please modify https://github.com/kubesphere/kubesphere/blob/master/.github/workflows/build-multiarch.yaml too and cherry-pick this pr

done

**Pull request merged**